azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -343,8 +329,35 @@ def _find_existing_spc(self, instance_name: str, cl_resources: List[dict]) -> di
 
         return secretsync_spc
 
-    def _check_secret_name(self, secrets: PageIterator, secret_name: str, spc_keyvault_name: str, flag: str) -> str:
+    def _find_resource_from_cl_resources(
+        self,
+        cl_resources: List[dict],
+        resource_type: str,
+        resource_name: Optional[str] = None,
+    ) -> dict:
+        for resource in cl_resources:
+            resource_id_container = parse_resource_id(resource["id"])
+            cl_resource_name = resource_id_container.resource_name
 
+            # Ensure both type and name (if specified) match the resource
+            is_name_matched = resource_name is None or cl_resource_name == resource_name
+            is_type_matched = resource["type"].lower() == resource_type
+
+            if is_type_matched and is_name_matched:
+                if resource_type == SPC_RESOURCE_TYPE:
```

Comment:
nit: would have been more generic + could have been put in the instance class or somewhere else if you took out the sdk calls but it is good enough for now

Reply:
ok I will add a todo for considering generalize it, thanks!